app/ide-desktop/lib/dashboard/src/components/AriaComponents/Form/components/types.ts

```diff
@@ -9,27 +9,38 @@ import type * as z from 'zod'
 /**
  * Field Values type.
  */
-export type FieldValues = reactHookForm.FieldValues
+export type FieldValues<Schema extends TSchema> = Schema extends never
```

Comment:
type does not work: [TS playground](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAYg9nAPAFQHxQLxWVCAPYCAOwBMBnKIiANwgCcoB+KAcgEsKra6WoAuVh0pxglGvRYAoSaEhQA6gAsAhqKzwkXeqkkB6XVEMA9RkA)

distributive conditional types needs to be disabled: [playground](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAYg9nAPAFQHxQLxQNrILpQQAewEAdgCYDOOZEAbhAE4ED8UA5AJY12NMcoALk48oZOMHENmHAFBzQkKAHUAFgEMpWeEj7NUcgPRGoZgHqsgA)

Reply:
From the user perspective, there's no difference, but okay, thanks for the suggestion! Do have an idea how to fix the issue with `defaultValues`? 

The Problem is if we have both `schema` and `defaultValues`, if value is not specified in `defaultValues`, it gets omitted from `FieldProps`.

Reply:
i guess the main issue is that RHF doesn't support `defaultValues` having a different type than `schema`?

you could use zod `.default()` instead of `defaultValues` but that would mean you can't unconditionally `partial()` the entire object

[defaults.patch.txt](https://github.com/enso-org/enso/files/15496063/defaults.patch.txt)

Reply:
AFAIR .default() doesn't work in schema.

Reply:
A solution might be to try to parse an empty object by the schema and pass it as defaultValues

Reply:
But this won't work with async default values 🫠